src/components/intent.ts

```diff
@@ -838,9 +874,10 @@ export class Intent {
      * Ensures that the client has the required power level to post the event type.
      * @param roomId Required as power levels exist inside a room.
      * @param eventTypes The event type to check the permissions for.
+     * @param isState Are we checking for state permissions or regular event permissions.
      * @return If found, the power level event
      */
-    private async _ensureHasPowerLevelFor(roomId: string, eventType: string) {
+    private async _ensureHasPowerLevelFor(roomId: string, eventType: string, isState: boolean) {
```

Comment:
slightly instinctively I don't like bool for isState. should this be an enum? Maybe not because it will only ever be two...

Reply:
I see what you mean, but for the foreseeable there will only be two kinds of events that power levels can control.